From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect sound structures
Patch-mainline: never, kabi

In 3.0.52, commit 40edba6679 (ALSA: Add a reference counter to card
instance), upstream commit a0830dbd4e, added refcount to struct
snd_card and card_ptr to struct snd_minor. It indeed changed layout of
those structures.

Let's ignore that as snd_minor is internal only and snd_card is
allocated in the sound core. The only thing we need to take care of is
to put refcount at the end of the structure, so that everybody sees
still at the same position of struct snd_card.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/sound/core.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/include/sound/core.h
+++ b/include/sound/core.h
@@ -132,7 +132,6 @@ struct snd_card {
 	int shutdown;			/* this card is going down */
 	int free_on_last_close;		/* free in context of file_release */
 	wait_queue_head_t shutdown_sleep;
-	atomic_t refcount;		/* refcount for disconnection */
 	struct device *dev;		/* device assigned to this card */
 	struct device *card_dev;	/* cardX object for sysfs */
 
@@ -146,6 +145,9 @@ struct snd_card {
 	struct snd_mixer_oss *mixer_oss;
 	int mixer_oss_change_count;
 #endif
+#ifndef __GENKSYMS__
+	atomic_t refcount;		/* refcount for disconnection */
+#endif
 };
 
 #ifdef CONFIG_PM
@@ -190,7 +192,9 @@ struct snd_minor {
 	const struct file_operations *f_ops;	/* file operations */
 	void *private_data;		/* private data for f_ops->open */
 	struct device *dev;		/* device for sysfs */
+#ifndef __GENKSYMS__
 	struct snd_card *card_ptr;	/* assigned card instance */
+#endif
 };
 
 /* return a device pointer linked to each sound device as a parent */
